ofono/include/sim.h

```diff
@@ -211,7 +215,9 @@ struct ofono_sim_driver {
 	void (*logical_access)(struct ofono_sim *sim, int session_id,
 			const unsigned char *pdu, unsigned int len,
 			ofono_sim_logical_access_cb_t cb, void *data);
-	/* Since mer/1.23+git28 */
+	void (*set_active_card_slot)(struct ofono_sim *sim, unsigned int index,
```

Comment:
At very least, new callbacks must be added to the end of `struct ofono_sim_driver`. Even that is dangerous, because older plugins will have a shorter function table which is not necessarily followed by zeros🤔

Reply:
Something like https://github.com/sailfishos/ofono/pull/31 may be required.

Reply:
@piggz you're not there yet but good habit is to resolve threads once you're done. This might be super clear for your -- just saying. Keep up with the good work! :+1:

Reply:
Thanks @rainemak .. for this thread, im not if it is resolved, see comment below ,,,, order has been changed, but not sure if something else is needed to change the code to be built on #31